pkg/controller/windowsmachineconfig/windows/windows.go

```diff
@@ -259,6 +265,96 @@ func (vm *Windows) ConfigureCNI(configFile string) error {
 	return nil
 }
 
+// service represents a Windows service
+type service struct {
+	// binPath includes a path to the binary and all arguments to run the binary with
+	binPath string
+	// name is the name of the service
+	name string
+	// vm is the VM the service will be started on
+	vm *Windows
+}
+
+// newService returns a pointer to a new service
+func newService(vm *Windows, name, binPath string) *service {
+	return &service{
+		vm:      vm,
+		name:    name,
+		binPath: binPath,
+	}
+}
+
+// start creates and starts the service on the VM
+func (s *service) start() error {
+	_, stderr, err := s.vm.Run("sc.exe create "+s.name+" binPath=\""+s.binPath+"\" start=auto", false)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create service with stderr '%s'", stderr)
+	}
+	_, stderr, err = s.vm.Run("sc.exe start "+s.name, false)
+	if err != nil {
+		return errors.Wrapf(err, "failed to start service with stderr '%s'", stderr)
+	}
+	log.V(1).Info("started service", "name", s.name, "binPath", s.binPath)
+	return nil
+}
+
+// remove removes the service from the VM
+func (s *service) remove() error {
+	_, stderr, err := s.vm.Run("sc.exe delete "+s.name, false)
+	// Exit code 1060 is due to the service not existing. In that case we can return successfully
+	if err != nil && !strings.Contains(err.Error(), "1060 exit code") {
+		return errors.Wrapf(err, "failed to remove service with stderr '%s'", stderr)
+	}
+	return nil
+}
+
+// StartKubeProxyService ensures that the kube-proxy service is running
+func (vm *Windows) StartKubeProxyService(nodeName, hostSubnet string) error {
```

Comment:
This function does more than just start the proxy service. Going with the pattern we have, I suggest `ConfigureKubeProxy`.